listenbrainz/webserver/static/js/src/LastFMImporter.tsx

```diff
@@ -60,14 +225,149 @@ export default class LastFmImporter extends React.Component<
     });
   };
 
-  setClose = () => {
-    this.setState({ canClose: this.importer.canClose });
+  setClose = (canClose: boolean) => {
+    this.setState({ canClose });
+  };
+
+  updateMessage = (msg: React.ReactElement) => {
+    this.setState({ msg });
   };
 
-  updateMessage = () => {
-    this.setState({ msg: this.importer.msg });
+  handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({ lastfmUsername: event.target.value });
+  };
+
+  handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    const { lastfmUsername } = this.state;
+    this.toggleModal();
+    event.preventDefault();
+    this.startImport();
   };
 
+  async submitPage(payload: Array<Listen>) {
+    const delay = this.getRateLimitDelay();
+    // Halt execution for some time
+    await new Promise((resolve) => {
+      setTimeout(resolve, delay);
+    });
+
+    const response = await this.APIService.submitListens(
+      this.userToken,
+      "import",
+      payload
+    );
+    this.updateRateLimitParameters(response);
+  }
+
+  async startImport() {
+    this.updateModalAction(<p>Your import from Last.fm is starting!</p>, false);
+    this.playCount = await this.getTotalNumberOfScrobbles();
```

Comment:
this playCount is used only when it's not an incremental import. Imo we can add a check to only fetch the count from last.fm if it's not an incremental import. Saves an extra API call :)

Reply:
Thanks for the review @shivam-kapila. This component need to fetch the api at this point.. otherwise it is showing modal with success at an invalid username too

Reply:
Hm. Makes sense